estuary_updater/handlers/freshmaker.py

```diff
@@ -79,30 +80,30 @@ def build_state_handler(self, msg):
         """
         build_info = msg['body']['msg']
         event_id = msg['body']['msg']['event_id']
+        freshmaker_build = self.create_or_update_freshmaker_build(build_info, event_id)
         build = self.create_or_update_build(build_info, event_id)
+        event = None
         if build:
             event = FreshmakerEvent.nodes.get_or_none(id_=str(event_id))
             if event:
-                event.triggered_container_builds.connect(build)
+                event.successful_koji_builds.connect(build)
             else:
                 log.warn('The Freshmaker event {0} does not exist in Neo4j'.format(event_id))
+        if freshmaker_build and event:
+            freshmaker_build.event.connect(event)
+            event.requested_builds.connect(freshmaker_build)
 
-    def create_or_update_build(self, build, event_id):
+    def create_or_update_freshmaker_build(self, build, event_id):
         """
-        Use the Koji Task Result to create or update a ContainerKojiBuild.
+        Create or update a FreshmakerBuild.
 
         :param dict build: the build represented in Freshmaker being created or updated
         :param int event_id: the id of the Freshmaker event
-        :return: the created/updated ContainerKojiBuild or None if it cannot be created
-        :rtype: ContainerKojiBuild or None
+        :return: the created/updated FreshmakerBuild or None if it cannot be created
+        :rtype: FreshmakerBuild or None
         """
-        # Builds in Koji only exist when the Koji task this Freshmaker build represents completes
-        if build['state'] != 1:
```

Comment:
You are right that FreshmakerBuild nodes should be created/updated for every state change, but the if statement should remain in `create_or_update_build` because the build in Koji only exists when it is in the done state (i.e. `koji.BUILD_STATES['COMPLETE']`).